pyrdp/security/nla.py

```diff
@@ -33,15 +35,35 @@ def __init__(self, sink: IntermediateLayer, state: NTLMSSPState, log: logging.Lo
         self.sink = sink
         self.ntlmSSPState = state
         self.log = log
+        self.catch = ntlmCatch
         self.ntlmSSPParser = NTLMSSPParser()
 
+    def getRandChallenge(self):
```

Comment:
Should we use the server-provided random values? I'm afraid we are going to introduce a vulnerability here by using a random number generator that is known to be weak: 
![image](https://user-images.githubusercontent.com/546325/142060741-94b5342a-4332-4686-91d4-938d57f0e217.png)

from https://docs.python.org/3/library/random.html

Reply:
The RDP server is the one in charge of creating the challenge value so yes, it should be created by PyRDP in this case. Either way, good catch with `random`. Pushed commit 1129bbb to use `secrets` instead.

Reply:
Just to make sure, we don't have a challenge we could use in the TSREQUEST[SPNEGO] from the server? This is when we drop the server's connection, we realize we can't MITM CredSSP so we drop. But we did get that SPNEGO, correct? Doesn't it hold the challenge already?

![image](https://user-images.githubusercontent.com/546325/142278138-05e7b468-9e23-45a8-a291-2e319aa273ba.png)

Reply:
No, the SPNEGO is sent in an already established NTLMSSP authentication mode. The connection with the server is cut way before any CredSSP is done.